src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -16,6 +16,14 @@ public class SocketTransportOptions
         /// </remarks>
         public int IOQueueCount { get; set; } = Math.Min(Environment.ProcessorCount, 16);
 
+        /// <summary>
+        /// Wait until there is data available to allocate a buffer. Setting this to false can increase throughput at the cost of increased memory usage.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
+        public bool WaitForDataBeforeAllocatingBuffer { get; set; } = false;
```

Comment:
Seems reasonable. I don't necessarily add a ton of knobs, but this one seems worthwhile. I'll approve once this gets switched back to true. I don't want to accidentally switch the default.